internal/osnadmin/fetch.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package osnadmin
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hyperledger/fabric-admin-sdk/internal/protoutil"
+	"github.com/hyperledger/fabric-admin-sdk/pkg/identity"
+	cb "github.com/hyperledger/fabric-protos-go-apiv2/common"
+	ab "github.com/hyperledger/fabric-protos-go-apiv2/orderer"
+)
+
+var seekNewest = &ab.SeekPosition{
+	Type: &ab.SeekPosition_Newest{
+		Newest: &ab.SeekNewest{},
+	},
+}
+
+func Fetch(service ab.AtomicBroadcast_DeliverClient,
```

Comment:
@bestbeforetoday , please let me know if gateway has interface to generate AtomicBroadcast_DeliverClient instance for fetch latest block function?

Reply:
Gateway uses peer.DeliverClient to obtain block events from the Gateway peer. The process is essentially the same so, while you could borrow from the Gateway code how to assemble the request message and read the resulting blocks, I don't think the Gateway API is going to be usable directly to read from the orderer.

The Gateway client API also only offers the options of specific start block number or next committed block number, both with no defined end block. For reading the config block from the orderer, you will instead want to read the last commit block, followed by reading the specific latest config block number. For efficiency, you might also want to specify and end block, rather than just closing the context at the client end once you have received the required block.

Within the Gateway code, the relevant pieces are:
1. [Build the event request](https://github.com/hyperledger/fabric-gateway/blob/b1659c8d92917b3237aaceb9f6d039c049c6897a/pkg/client/blockeventsbuilder.go#L118-L134)
1. [Establish the event stream and send the event request](https://github.com/hyperledger/fabric-gateway/blob/b1659c8d92917b3237aaceb9f6d039c049c6897a/pkg/client/client.go#L97-L108)
1. [Read blocks from the returned stream event client](https://github.com/hyperledger/fabric-gateway/blob/b1659c8d92917b3237aaceb9f6d039c049c6897a/pkg/client/blockevents.go#L109-L127)

Your reading code is simpler since you just read a single block for each of your two requests, then close the Context.